### ISSUE TITLE
32 character limit for names should probably be 32 codepoints

### ISSUE THREAD

**Issue by ToxicMushroom** [DV8FromTheWorld/JDA#1556]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

## Bug Report

Im able to configure my name on dicord like this: `𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸`
Which looks like 32 characters but with utf-8 this is actually 64 chars in length.
It still is 32 codepoints but jda checks the character count in some functions instead.
`JDA#getUserByTag(String username, String discriminator)` for example

### Expected Behavior
JDA should not throw an error when trying to supply usernames that fit into discord.
I doubt discord will change this behaviour so we should support it perhaps.

You could count codepoints on the string with `String#codePoints().count()`
But I'm unsure if this would allow inputs that would be too big for discord.

### Code Example or Reproduction Steps
```kotlin
shardManager.getUserByTag("𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸𝐸𝐿𝐼𝒮𝐸#5602")
```

### Code for JDABuilder or DefaultShardManagerBuilder Used
N/A

### Exception or Error
```
java.lang.IllegalArgumentException: Username must be between 2 and 32 characters in length!
	at net.dv8tion.jda.internal.utils.Checks.check(Checks.java:43)
	at net.dv8tion.jda.api.JDA.getUserByTag(JDA.java:666)
	at net.dv8tion.jda.api.JDA.getUserByTag(JDA.java:635)
	at net.dv8tion.jda.api.sharding.ShardManager.lambda$getUserByTag$12(ShardManager.java:612)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at net.dv8tion.jda.api.sharding.ShardManager.lambda$getUserByTag$13(ShardManager.java:614)
	at net.dv8tion.jda.api.utils.cache.CacheView.applyStream(CacheView.java:151)
	at net.dv8tion.jda.api.sharding.ShardManager.getUserByTag(ShardManager.java:611)
	at me.melijn.melijnbot.internals.utils.JDAUtilsKt.getUserByArgsN(JDAUtils.kt:137)
	at me.melijn.melijnbot.internals.utils.JDAUtilsKt.getUserByArgsN(JDAUtils.kt:124)
	at me.melijn.melijnbot.internals.utils.JDAUtilsKt.retrieveUserByArgsN(JDAUtils.kt:154)
	at me.melijn.melijnbot.internals.utils.JDAUtilsKt.retrieveUserByArgsNMessage(JDAUtils.kt:198)
	at me.melijn.melijnbot.commands.utility.UserInfoCommand.execute(UserInfoCommand.kt:29)
	at me.melijn.melijnbot.internals.command.AbstractCommand.run(AbstractCommand.kt:156)
	at me.melijn.melijnbot.internals.command.AbstractCommand$run$1.invokeSuspend(AbstractCommand.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1429)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)```

**Comment by Andre601** [comment]
> JDA should not throw an error when trying to supply usernames that fit into discord.
I doubt discord will change this behaviour so we should support it perhaps.

Discord in fact supports this. The button to change your name will turn into a face if you use those unicode text symbols.
Just an FYI.

![image](https://cdn.discordapp.com/attachments/367465113172705282/807985088045383720/unknown.png)

**Comment by ToxicMushroom** [comment]
this issue was fixed by #1579
